src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -308,6 +308,149 @@ public TimeSpan Duration()
 
         public override int GetHashCode() => _ticks.GetHashCode();
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TimeSpan"/> structure to a specified number of
+        /// days.
+        /// </summary>
+        /// <param name="days">Number of days.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// The parameters specify a <see cref="TimeSpan"/> value less than <see cref="MinValue"/> or greater than <see cref="MaxValue"/>
+        /// </exception>
+        public static TimeSpan FromDays(int days) => FromDays(days, 0);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TimeSpan"/> structure to a specified number of
+        /// days, hours, minutes, seconds, milliseconds, and microseconds.
+        /// </summary>
+        /// <param name="days">Number of days.</param>
+        /// <param name="hours">Number of hours.</param>
+        /// <param name="minutes">Number of minutes.</param>
+        /// <param name="seconds">Number of seconds.</param>
+        /// <param name="milliseconds">Number of milliseconds.</param>
+        /// <param name="microseconds">Number of microseconds.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// The parameters specify a <see cref="TimeSpan"/> value less than <see cref="MinValue"/> or greater than <see cref="MaxValue"/>
+        /// </exception>
+        public static TimeSpan FromDays(int days, int hours = 0, long minutes = 0, long seconds = 0, long milliseconds = 0, long microseconds = 0)
+        {
+            Int128 totalMicroseconds = Math.BigMul(days, MicrosecondsPerDay)
+                                     + Math.BigMul(hours, MicrosecondsPerHour)
+                                     + Math.BigMul(minutes, MicrosecondsPerMinute)
+                                     + Math.BigMul(seconds, MicrosecondsPerSecond)
+                                     + Math.BigMul(milliseconds, MicrosecondsPerMillisecond)
+                                     + microseconds;
+
+            if ((totalMicroseconds > MaxMicroseconds) || (totalMicroseconds < MinMicroseconds))
```

Comment:
>if ((totalMicroseconds > MaxMicroseconds) || (totalMicroseconds < MinMicroseconds)) [](http://example.com/codeflow?start=12&length=83)

I think this is correct, I am just wondering why @tannergooding suggested doing like 

`if (((upperMicroseconds == 0) && ((ulong)lowerMicroseconds > MaxMicroseconds)) || (upperMicroseconds != -1) || ((ulong)lowerMicroseconds < (ulong)MinMicroseconds))`

Reply:
I kept the original comparison here because it works.
When I tried to simply copy Tanners suggestion it would not build, and my attempts at corrections made the tests fail.
To go further with that I think I would have to spend some time playing with `Int128`, `long` and `ulong` to get from "understand the general idea of what Tanner is trying to do" to actually "understand how to make that work".

I assume Tanner suggested this for performance, though I don't know how this helps.
I'm guessing comparison to Int128 has some overhead that may make it worth doing more and more complicated comparisons.

I also tried to use `dotnet/performance` repo to benchmark the actual implementation, but I could not get the benchmark to find the new overloads.
If I understood the docs right, that should be a lot easier shortly after they are merged.

PS: I expect to do a follow up PR to do the trivial optimizations of the single argument methods.